File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/colors/ChemtechColor.java
package com.bonepl.chromaleague.hud.colors;

import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.color.MultiTransitionColor;
import com.bonepl.razersdk.color.StaticColor;

import java.security.SecureRandom;

public class ChemtechColor implements Color {
    final MultiTransitionColor chemtechColor = new MultiTransitionColor.Builder(CLColor.CHEMTECH)
            .addTransition(StaticColor.YELLOW, 10)
            .addTransition(CLColor.CHEMTECH, 10)
            .addTransition(new StaticColor(25, 60, 0), 5)
            .looped(5)
            .build();

    public ChemtechColor() {
        int random = new SecureRandom().nextInt(chemtechColor.getTotalTransitions());
        for (int i = 0; i < random; i++) {
            chemtechColor.getColor();
        }
    }

    @Override
    public StaticColor getColor() {
        return chemtechColor.getColor();
    }
}

File: src/hud/colors/ChemtechColor.ts
import {Color} from '../../razer-sdk/color/Color.js';
import {MultiTransitionColor} from '../../razer-sdk/color/MultiTransitionColor.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {CLColor} from './CLColor.js';

export class ChemtechColor implements Color {
    readonly chemtechColor = new MultiTransitionColor.Builder(CLColor.CHEMTECH)
        .addTransition(StaticColor.YELLOW, 10)
        .addTransition(CLColor.CHEMTECH, 10)
        .addTransition(new StaticColor(25, 60, 0), 5)
        .looped(5)
        .build();

    constructor() {
        const random = Math.trunc(Math.random() * this.chemtechColor.getTotalTransitions());
        for (let i = 0; i < random; i++) {
            this.chemtechColor.getColor();
        }
    }

    getColor(): StaticColor {
        return this.chemtechColor.getColor();
    }
}
